### DISCUSSION TITLE
Change style of Custom component from another custom component with variant in theme.

### DISCUSSION THREAD

**Discussion by ankursehdev** [chakra-ui/chakra-ui#7198] [Q&A] [answered]
### Description

Hello, I am working on a big project and creating our own design system. I will try to explain the situation here.

**I have a multipart generic CARD component with 2 variants. I want this to define structure only.* and user can use this to create other components*

**The idea is this can be used in a lot of places and I don't want to control the design here.** 
This work well. something similar to this.

```
const Card = {
parts: [Card, CardMedia, CardContent],
  baseStyle: {
    display: 'flex',
    flexDirection: 'column',
  },
  variants: {
    vertical: {
      //styles here
    },
    horizontal: {
     //styles here
    },
  },
  defaultProps: {
    variant: 'collections',
  },
}
```

### Problem Statement/Justification

Now, there is a different **Section** that has multiple variants as well- I need to use existing CARD in this but want different styles applied to it.

In the Section, There is a different requirement of the design. 
Image has some border
Card has different Padding, margins etc
CardContent needs hover or other things. 

**I need to change some styles of Card, CardMedia, and CardContent. Can I do it through the theme?**

I am attaching the designs also for reference. These all are kinds of cards with different styles. 

If you look below all of these have an 
```
<Card>
IMAGE <MEDIA>
CONTENT <CardContent>
</Card>
```
1. I want to make a generic Card component, so it can be used everywhere.
2. Use this card component in other components to create different styles with props.

![Screen Shot 2023-01-05 at 2 45 13 PM](https://user-images.githubusercontent.com/34501847/210867113-cdf02787-7f40-444e-b920-211b02b6e713.png)
![Screen Shot 2023-01-05 at 2 45 15 PM](https://user-images.githubusercontent.com/34501847/210867114-959478b3-d9bc-41fc-a11b-1d0d0691bf4c.png)
![Screen Shot 2023-01-05 at 2 45 20 PM](https://user-images.githubusercontent.com/34501847/210867115-aecab6f4-212f-4682-9adb-82647678a06f.png)
![Screen Shot 2023-01-05 at 2 45 34 PM](https://user-images.githubusercontent.com/34501847/210867117-9929c83c-5d5c-4ee8-ba0e-7cd7f86a0123.png)

Please guide me if this is possible. Currently, I am unable to do this with this approach.

TIA

### Proposed Solution or API

I am trying to do something like this: 

```
const BannerSection = defineStyleConfig({
  baseStyle: {
    display: 'flex',
    flexDirection: 'column',
  },
  // Two variants: rounded and smooth
  variants: {
    inline: {
      /** CAN I REFER TO CARD, CARDMEDIA, CARDCONTENT HERE SOMEHOW? **/
      cardMedia: {
        //Some custom styles here for section only. 
      },
    },
    full: {
      /** CAN I REFER TO CARD, CARDMEDIA, CARDCONTENT HERE SOMEHOW? something like**/
      card: {
        border: `1px solid green`,
      },
    },
  },
  defaultProps: {
    variant: 'full',
  },
});
```

**Comment by segunadebayo** [comment, marked as answer]
Hi @ankursehdev,

You can always add a className to each component and refer to them in the styles.

You should make the component composable instead of trying to solve them with a single theme definition (this will lead to unmaintainable styles).

**Reply by ankursehdev** [reply]
Thanks for the context, this makes sense.
